graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/AllKeyspacesFetcher.java

```diff
@@ -17,21 +17,46 @@
 
 import graphql.schema.DataFetchingEnvironment;
 import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthorizationService;
 import io.stargate.db.Persistence;
 import io.stargate.db.datastore.DataStore;
 import io.stargate.graphql.schema.fetchers.CassandraFetcher;
+import io.stargate.graphql.web.HttpAwareContext;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class AllKeyspacesFetcher extends CassandraFetcher<List<Map<String, Object>>> {
 
-  public AllKeyspacesFetcher(Persistence persistence, AuthenticationService authenticationService) {
-    super(persistence, authenticationService);
+  public AllKeyspacesFetcher(
+      Persistence persistence,
+      AuthenticationService authenticationService,
+      AuthorizationService authorizationService) {
+    super(persistence, authenticationService, authorizationService);
   }
 
   @Override
-  protected List<Map<String, Object>> get(
-      DataFetchingEnvironment environment, DataStore dataStore) {
-    return KeyspaceFormatter.formatResult(dataStore.schema().keyspaces(), environment);
+  protected List<Map<String, Object>> get(DataFetchingEnvironment environment, DataStore dataStore)
+      throws Exception {
+    HttpAwareContext httpAwareContext = environment.getContext();
+    String token = httpAwareContext.getAuthToken();
+
+    List<Map<String, Object>> keyspaces =
+        KeyspaceFormatter.formatResult(dataStore.schema().keyspaces(), environment);
+    for (Map<String, Object> keyspace : keyspaces) {
+      String keyspaceName = (String) keyspace.get("name");
+      @SuppressWarnings("unchecked")
+      List<Map<String, Object>> tables = (List<Map<String, Object>>) keyspace.get("tables");
+
+      List<String> tableNames = null;
+      if (tables != null) {
+        tableNames = tables.stream().map(t -> (String) t.get("name")).collect(Collectors.toList());
+      }
+
+      authorizationService.authorizeSchemaRead(
+          token, Collections.singletonList(keyspaceName), tableNames);
```

Comment:
I think it would be better to push that logic into `KeyspaceFormatter`: check as we go, instead of parsing everything and checking after the fact. It will also allow code reuse between the single-keyspace and multi-keyspace methods.